lib/bolt/cli.rb

```diff
@@ -109,7 +109,8 @@ def create_option_parser(results)
           '* protocol is `ssh` by default, may be `ssh` or `winrm`',
           '* port is `22` by default for SSH, `5985` for winrm (Optional)'
         ) do |nodes|
-          results[:nodes] = parse_nodes(nodes)
+          results[:nodes] = [] unless results[:nodes]
+          results[:nodes] = results[:nodes] + parse_nodes(nodes)
```

Comment:
This could cause a regression on `uniq`ness as implemented in https://github.com/puppetlabs/bolt/pull/96.

A more idiomatic ruby would be to:

```
parsed = parse_nodes(nodes)
if parsed
  results[:nodes] += parsed
else
  results[:nodes] = []
end
```

Reply:
The problem would be you are assigning the result a nil list if not parsed again, so if it fails the last parse it will than give an empty list?

Reply:
Also at the point of parsing the results[:nodes] would be nil

Reply:
Also now that I am looking more, any reason not to switch to a struct style options parser pattern?